CRM/Civicase/Api/Wrapper/CaseGetList.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+use CRM_Case_BAO_CaseType as CaseType;
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+
+/**
+ * CRM_Civicase_ApiWrapper_CaseGetList.
+ */
+class CRM_Civicase_Api_Wrapper_CaseGetList implements API_Wrapper {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function fromApiInput($apiRequest) {
+    if (!$this->canHandleTheRequest($apiRequest)) {
+      return $apiRequest;
+    }
+
+    $accessibleCaseCategories = CaseCategoryHelper::getAccessibleCaseTypeCategories();
+    $caseCategoryParameterValue = $this->getCaseTypeCategoryParameterValue($apiRequest['params']);
+
+    if (!empty($caseCategoryParameterValue)) {
+      $this->validateCaseTypeCategoryParameter($caseCategoryParameterValue, $accessibleCaseCategories);
+    }
+    else {
+      $apiRequest['params']['params']['case_id.case_type_id.case_type_category'] = ['IN' => $accessibleCaseCategories];
+    }
+
+    return $apiRequest;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function toApiOutput($apiRequest, $result) {
+
+    return $result;
+  }
+
+  /**
+   * Checks whether the request can be handled or not.
```

Comment:
Can we describe when the request can be handled instead? Like we only accept requests coming from the `Case.getlist` endpoint.